lldb/tools/lldb-dap/Handler/DisassembleRequestHandler.cpp

```diff
@@ -56,100 +54,204 @@ DisassembleRequestHandler::Run(const DisassembleArguments &args) const {
     }
   }
 
+  int64_t instructionOffset = args.instructionOffset.value_or(0);
+  if (instructionOffset > 0) {
+    lldb::SBInstructionList forward_insts = dap.target.ReadInstructions(
+        addr, instructionOffset + 1, flavor_string.c_str());
+    if (forward_insts.GetSize() != static_cast<size_t>(instructionOffset + 1)) {
+      return llvm::make_error<DAPError>(
+          "Failed to disassemble instructions after " +
+          std::to_string(instructionOffset) +
+          " instructions from the given address.");
```

Comment:
Just to make sure I understand: we're erroring out here because if there are already not enough instructions to cover the instruction offset, there's no way we can assemble anymore (i.e. the ones covered by instructionCount)? Might be worth a comment. My initial reaction was: why not do a best case effort.